platform/src/main/resources/config/anchor-config-default-values.yaml

```diff
@@ -681,15 +681,22 @@ events:
       poll_timeout_seconds: 60
       # The security protocol used to communicate with the brokers.
       # `security_protocol` can be one of the following:
-      #     `PLAINTEXT`: no encryption
-      #     `SASL_PLAINTEXT`: SASL authentication with no encryption
+      #   `PLAINTEXT`: no encryption
+      #   `SASL_PLAINTEXT`: SASL authentication with no encryption
+      #   `SASL_SSL`: SASL authentication with SSL encryption
       # If SASL_PLAINTEXT is used, the SECRET_EVENTS_QUEUE_KAFKA_USERNAME and SECRET_EVENTS_QUEUE_KAFKA_PASSWORD
       # environment variables must be set.
       security_protocol: PLAINTEXT
-      # The SASL mechanism used for authentication when `security_protocol` is set to `SASL_PLAINTEXT`.
-      # `sasl_mechanism` can be one of the following:
-      #     `PLAIN`: PLAIN SASL mechanism
-      sasl_mechanism:
+      # The SASL mechanism used for authentication and must match what is used on the broker.
+      # `sasl_mechanism` can be one of the following supported mechanisms:
+      #   `PLAIN`: PLAIN SASL mechanism
+      #   `SCRAM-SHA-256`: SCRAM-SHA-256 SASL mechanism
+      #   `SCRAM-SHA-512`: SCRAM-SHA-512 SASL mechanism
+      #   `GSSAPI`: GSSAPI SASL mechanism
+      #   `OAUTHBEARER`: OAUTHBEARER SASL mechanism
+      # ` AWS_MSK_IAM`: AWS MSK IAM SASL mechanism
```

Comment:
Do we support these?

Reply:
The values are defined in the enum. We don't have to support the implementation. The Kafka client library implements these mechanisms.

Reply:
IAM-specific auth methods need additional properties configured https://github.com/aws/aws-msk-iam-auth?tab=readme-ov-file#configuring-a-kafka-client-to-use-aws-iam-with-aws_msk_iam-mechanism

There may be specific properties additionally required for the other SASL mechanisms.

Reply:
Keep only `PLAIN`